design-proposals/toggle_vga_video_device.md

```diff
@@ -0,0 +1,96 @@
+# Overview
+We want to allow cluster administrators and VM owners to toggle the video device type.
+between VGA (for compatibility with older VMs) and virtio (that will be used by default).
+
+## Motivation
+Currently, the default video device type for AMD64 in KubeVirt is VGA.
+Which limits the functionality available to virtual machines.
+For example, VGA restricts display resolution settings for Windows VMs and may not support modern workloads effectively.
+However, it is necessary to maintain compatibility for older virtual machines that depend on VGA (e.g., RHEL 5/6).
+To address this, we propose making virtio the default video device type for AMD64 systems while providing a mechanism to toggle VGA for older VMs.
+
+## Goals
+the kubevirt resource should have the ability to toggle the default video device between vga and virtio
+
+## Non Goals
+(limitations to the scope of the design)
+
+## Definition of Users
+Cluster admins managing KubeVirt deployments who need to standardize the default video device type across their cluster.
+vm-owners who require specific default configurations for VMIs in their environments.
+
+## User Stories
+- As a cluster administrator, I want to ensure my old VMs continue using VGA as the default video device type
+while transitioning new workloads to use virtio by default.
+
+- As a VM owner, I want to ensure compatibility with older VMs that require VGA by marking the kv resource accordingly before upgrading.
+
+## Repos
+- [KubeVirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+
+### Option 1: Annotation-Based Toggle
+**Description:** Introduce an annotation on the KubeVirt CR to specify the default video device type. The annotation kubevirt.io/default-video-device: vga can be used to default all VMs to VGA unless overridden in individual VMIs.
+
+**Pros:**
+
+- Minimal changes to the API schema.
+- Easy to implement and maintain.
```

Comment:
TBH I don't understand these pros.

Such an annotation would demand backward compatibility anyhow, and it's not much easier to implement AFAICT. To me it's equivalent to adding a non-visible field.